src/pcre2test.c

```diff
@@ -3925,6 +3934,21 @@ for (;;)
       else *((uint32_t *)field) |= m->value;
     break;
 
+    case MOD_OPTMZ:
```

Comment:
FYI there is a macro to help with these code, but agree is not needed if it is only called once (which I was hoping would change, though)

Reply:
Hmm, I looked at the code again and it still seems that my first reading was correct.

`PUBLIC_COMPILE_OPTIONS` lists all the bits which can normally be set in `options`; but if 'PCRE2_LITERAL` is set, then only the bits in `PUBLIC_LITERAL_COMPILE_OPTIONS` can be set, or else compilation fails with an error. `PCRE2_NO_AUTO_POSSESS` and `PCRE2_NO_DOTSTAR_ANCHOR` are not in `PUBLIC_LITERAL_COMPILE_OPTIONS`, so those two options cannot be used to disable optimizations for literal patterns.

Doubtless this is because the 2 optimizations concerned do not apply to literal patterns anyways, so it doesn't make much sense to specifically disable them in that case.

Again, this raises the question: should we block users from enabling/disabling certain optimizations in the compile context for literal patterns? And again, my opinion is: no, I don't think there is a good reason to do that.

One reason is because the compile context can be re-used for multiple calls to `pcre2_compile()`. It might happen that some of those calls are for literal patterns, while others are not.

So again, from my viewpoint, this is a *desired* difference in behavior  between the legacy option bits and the new API function. However, if other key contributors agree that the use of `pcre2_set_optimize` should be restricted if the compile context is subsequently used for a literal pattern... well and good, I can implement it in that case.